### ISSUE TITLE
mypy --strict doesn't handle wildcard imports of modules that do 'import foo as bar'

### ISSUE THREAD

**Issue by jgarvin** [python/mypy#17267]
**To Reproduce**

a.py:
```python
import logging as log
```

b.py
```python
from a import *
log.info("Hello")
```

**Expected Behavior**

`b.py` should typecheck without any issue. At runtime it works fine and there are no runtime type errors, only mypy complains.

**Actual Behavior**

```
$ python3 -m mypy --strict /tmp/b.py
b.py:2: error: Name "log" is not defined  [name-defined]
Found 1 error in 1 file (checked 1 source file)
```

**Your Environment**

- Mypy version used: 1.10.0
- Mypy command-line flags: `--strict`
- Mypy configuration options from `mypy.ini` (and other config files): None
- Python version used: 3.12.3

**Comment by jgarvin** [comment]
Note that mypy does correctly handle the `as` creating a name within `a.py` if you're using the `from foo import bar as buzz` form then mypy correctly understands this creates the name `buzz` in `a.py`.

**Comment by jgarvin** [comment]
It also seems to work if you don't change the name, e.g. if `a.py` does `import os as os` then `b.py` can use `os.remove` just fine. Weird.

**Comment by hauntsaninja** [comment]
Yes, this should be related to the `--no-implicit-reexport` setting. A PR that improved the error message here to mention this would be useful.

**Comment by jgarvin** [comment]
It seems like in both the case that works and the case that doesn't an implicit re-export is happening though?
